drivers/modem/gsm_ppp.c

```diff
@@ -1172,13 +1229,32 @@ void gsm_ppp_stop(const struct device *dev)
 	net_if_l2(iface)->enable(iface, false);
 
 	if (IS_ENABLED(CONFIG_GSM_MUX)) {
+		gsm_soft_reset(gsm);
```

Comment:
If the mux is enabled we have AT channel access we can perform modem reset directly. After the soft reset muxing on modem side side is disabled and we disable muxing here too.